src/lib.rs

```diff
@@ -1002,12 +1052,40 @@ mod test {
     too_small!(small_int_6, 6, read_int);
     too_small!(small_int_7, 7, read_int);
 
+    too_small!(small_float_1, 1, read_float);
+    too_small!(small_float_2, 2, read_float);
+    too_small!(small_float_3, 3, read_float);
+    too_small!(small_float_4, 4, read_float);
+    too_small!(small_float_5, 5, read_float);
+    too_small!(small_float_6, 6, read_float);
+    too_small!(small_float_7, 7, read_float);
+
     #[test]
     fn uint_bigger_buffer() {
         use {ByteOrder, LittleEndian};
         let n = LittleEndian::read_uint(&[1, 2, 3, 4, 5, 6, 7, 8], 5);
         assert_eq!(n, 0x0504030201);
     }
+
+    // TODO: How is transmute implemented? Does it count as an operation on the sNaN?
+    //       Is this test undefined behavior?
```

Comment:
By the time you call transmute, I think you've already convert the sNaN to a qNaN, so I think you're safe.

(Even so, since it's only a test, I'm fine with being in murky territory.)